File: remote_switch/vbnet/RemoteSwitch.vb
Imports Tinkerforge

Module ExampleSimple
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "ctG" ' Change to your UID
	Const VALUE_A_ON  As Integer = (1 << 0) Or (1 << 2) ' Pin 0 and 2 high
	Const VALUE_A_OFF As Integer = (1 << 0) Or (1 << 3) ' Pin 0 and 3 high
	Const VALUE_B_ON  As Integer = (1 << 1) Or (1 << 2) ' Pin 1 and 2 high
	Const VALUE_B_OFF As Integer = (1 << 1) Or (1 << 3) ' Pin 1 and 3 high

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim iqr As New BrickletIndustrialQuadRelay(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

		iqr.SetMonoflop(VALUE_A_ON, 255, 1500) ' Set pins to high for 1.5 seconds
    End Sub
End Module

File: remote_switch/delphi/RemoteSwitch.pas
program RemoteSwitch;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletIndustrialQuadRelay;

type
  TRemoteSwitch = class
  private
    ipcon: TIPConnection;
    iqr: TBrickletIndustrialQuadRelay;
  public
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'ctG'; { Change to your UID }
  VALUE_A_ON  = (1 shl 0) or (1 shl 2); { Pin 0 and 2 high }
  VALUE_A_OFF = (1 shl 0) or (1 shl 3); { Pin 0 and 3 high }
  VALUE_B_ON  = (1 shl 1) or (1 shl 2); { Pin 1 and 2 high }
  VALUE_B_OFF = (1 shl 1) or (1 shl 3); { Pin 1 and 3 high }

var
  rs: TRemoteSwitch;

procedure TRemoteSwitch.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  iqr := TBrickletIndustrialQuadRelay.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Set pins to high for 1.5 seconds }
  iqr.SetMonoflop(VALUE_A_ON, 15, 1500);
end;

begin
  rs := TRemoteSwitch.Create;
  rs.Execute;
  rs.Destroy;
end.
